cmake/catkin_package.cmake

```diff
@@ -432,6 +432,31 @@ function(_catkin_package)
   # package cmake dir is the folder where the generated pkgConfig.cmake is located
   set(PKG_CMAKE_DIR "\${${PROJECT_NAME}_DIR}")
 
+  # strip SYSROOT prefixes for each element in LIST
+  macro(catkin_strip_sysroot)
```

Comment:
This should be a function instead of a macro. The only variable being changed in the parent scope should be the passed in list. That can easily be achieved with `set(... PARENT_SCOPE)`.

Why use two different terms? The function could be named `catkin_strip_root_path` to match what it does.

The function should also be moved to its own file and contain a docblock.